packages/protocol/contracts/signal/SignalService.sol

```diff
@@ -117,36 +117,33 @@ contract SignalService is AuthorizableContract, ISignalService {
             return false;
         }
 
-        bytes32 signalRoot = ICrossChainSync(p.crossChainSync).getSyncedSnippet(p.height).signalRoot;
+        bytes32 stateRoot = ICrossChainSync(p.crossChainSync).getSyncedSnippet(p.height).stateRoot;
 
-        if (signalRoot == 0) return false;
+        if (stateRoot == 0) return false;
 
         for (uint256 i; i < p.hops.length; ++i) {
             Hop memory hop = p.hops[i];
 
-            bytes32 label = authorizedAddresses[hop.signalRootRelay];
+            bytes32 label = authorizedAddresses[hop.signalService];
             if (label == 0) return false;
             uint64 chainId = uint256(label).toUint64();
 
             bytes32 slot = getSignalSlot(
                 chainId, // use label as chainId
-                hop.signalRootRelay,
-                hop.signalRoot // as a signal
+                hop.signalService,
+                hop.stateRoot // as a signal
             );
 
             bool verified = SecureMerkleTrie.verifyInclusionProof(
-                bytes.concat(slot), hex"01", hop.storageProof, signalRoot
+                bytes.concat(slot), hex"01", hop.storageProof, stateRoot
```

Comment:
Here we'll also have to prove the signal root account state root (aka the signal root). But we can cache it so only the first user that wants to use this state root has to do it. 

```
mapping(address signalService => mapping(bytes32 stateRoot => bytes32 signalRoot)) public cachedSignalRoots;

if (cachedSignalRoots[hop.signalService][stateRoot] == bytes32(0)) {
  bool accountVerified = SecureMerkleTrie.verifyInclusionProof(
      bytes.concat(hop.signalService), signalRoot, _transcode(hop.accountProof), stateRoot
  );
  if (!accountVerified) return false;

  // Cache for reuse for future users so they don't have to verify the account proof anymore
  cachedSignalRoots[hop.signalService][stateRoot] = signalRoot;
}

bool verified = SecureMerkleTrie.verifyInclusionProof(
    bytes.concat(slot), hex"01", _transcode(hop.storageProof), cachedSignalRoots[hop.signalService][stateRoot]
);
if (!verified) return false;
```